### PR TITLE
Retry persist

### PR DESCRIPTION
**Pull request by llanginger**
Adds retry functionality for failed persist calls.

### CODE REVIEW

File: src/wda/persist.js [hunk 1]
```diff
@@ -206,6 +212,19 @@ async function persist(input) {
 		})
 }
 
+// Retry failed attempts to persist:
+async function retry(fn, retriesLeft = 3, interval = 1000, exponential = false) {
```
Comment (dlewissandy) [inline src/wda/persist.js:216]: The interval is probably too short for persistence queries.    I would use a default of 10 seconds.

**Review by dlewissandy — APPROVED** [review]

File: src/wda/persist.js [hunk 2]
```diff
@@ -206,6 +212,19 @@ async function persist(input) {
 		})
 }
 
+// Retry failed attempts to persist:
```
Comment (llanginger) [inline src/wda/persist.js:215]: Increased timeout to 10000

File: src/wda/persist.js [hunk 3]
```diff
@@ -206,6 +212,19 @@ async function persist(input) {
...
+    const val = await fn();
+    return val;
+  } catch (error) {
+    if (retriesLeft) {
+      await new Promise(r => setTimeout(r, interval));
+      return retry(fn, retriesLeft - 1, exponential ? interval * 2 : interval, exponential);
+    } else throw new Error(`Max retries reached with error: ${error.message}`);
+  }
+}
+
```
Comment (llanginger) [inline src/wda/persist.js:227]: Added missing input to exported function

File: src/wda/persist.js [hunk 4]
```diff
@@ -206,6 +212,19 @@ async function persist(input) {
 		})
 }
 
+// Retry failed attempts to persist:
+async function retry(fn, retriesLeft = 3, interval = 10000, exponential = false) {
```
Comment (llanginger) [inline src/wda/persist.js:216]: Increased default timeout to 10000

File: src/wda/persist.js [hunk 5]
```diff
@@ -206,6 +212,19 @@ async function persist(input) {
...
+    if (retriesLeft) {
+      await new Promise(r => setTimeout(r, interval));
+      return retry(fn, retriesLeft - 1, exponential ? interval * 2 : interval, exponential);
+    } else throw new Error(`Max retries reached with error: ${error.message}`);
+  }
+}
+
 module.exports = {
-	persist,
+	persist: retry((input) => persist(input), input.retries, input.interval, input.exponentialRetries),
```
Comment (llanginger) [inline src/wda/persist.js:229]: Added missing input to exported function

**Review by dlewissandy — APPROVED** [review]

**Pull request merged**